src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibilityHelp.ts

```diff
@@ -3,80 +3,72 @@
+import { TerminalAccessibleWidget } from 'vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleWidget';
+import type { Terminal } from 'xterm';
+
+export const enum ClassName {
+	AccessibleBuffer = 'terminal-accessibility-help',
+	Active = 'active',
+	EditorTextArea = 'textarea'
+}
+
+export class AccessibilityHelpWidget extends TerminalAccessibleWidget {
 
-export class AccessibilityHelpWidget extends Widget {
-	readonly id = 'help';
-	private _domNode: FastDomNode<HTMLElement>;
-	get element(): HTMLElement { return this._domNode.domNode; }
-	private _contentDomNode: FastDomNode<HTMLElement>;
 	private readonly _hasShellIntegration: boolean;
-	private readonly _markdownRenderer: MarkdownRenderer;
 
 	constructor(
-		private readonly _instance: ITerminalInstance,
+		_instance: Pick<ITerminalInstance, 'shellType' | 'capabilities' | 'onDidRequestFocus' | 'resource'>,
+		_xterm: Pick<IXtermTerminal, 'getFont' | 'shellIntegration'> & { raw: Terminal },
 		@IKeybindingService private readonly _keybindingService: IKeybindingService,
-		@IInstantiationService instantiationService: IInstantiationService,
-		@IOpenerService private readonly _openerService: IOpenerService,
-		@IContextKeyService private readonly _contextKeyService: IContextKeyService
+		@IInstantiationService _instantiationService: IInstantiationService,
+		@IModelService _modelService: IModelService,
+		@IConfigurationService _configurationService: IConfigurationService,
+		@IAccessibilityService private readonly _accessibilityService: IAccessibilityService,
+		@IContextKeyService _contextKeyService: IContextKeyService,
+		@ITerminalService _terminalService: ITerminalService,
 	) {
-		super();
-		this._hasShellIntegration = _instance.xterm?.shellIntegration.status === ShellIntegrationStatus.VSCode;
-		this._domNode = createFastDomNode(document.createElement('div'));
-		this._contentDomNode = createFastDomNode(document.createElement('div'));
-		this._contentDomNode.setClassName('terminal-accessibility-help');
-		this._contentDomNode.setAttribute('role', 'document');
-		this._domNode.setDisplay('none');
-		this._domNode.setAttribute('role', 'dialog');
-		this._domNode.setAttribute('aria-hidden', 'true');
-		this._domNode.appendChild(this._contentDomNode);
-		this._register(addStandardDisposableListener(this._contentDomNode.domNode, 'keydown', (e) => {
-			if (e.keyCode === KeyCode.Escape) {
-				this.hide();
-				_instance.focus();
+		super(ClassName.AccessibleBuffer, _instance, _xterm, undefined, _instantiationService, _modelService, _configurationService, _contextKeyService, _terminalService);
+		this._hasShellIntegration = _xterm.shellIntegration.status === ShellIntegrationStatus.VSCode;
+		this.element.ariaRoleDescription = localize('terminal.integrated.accessiblityHelp', 'Terminal accessibility help');
+		this.add(once(this.editorWidget.onDidFocusEditorText)(() => {
+			// prevents tabbing into the editor
```

Comment:
we don't want tab focus to move into the help widget, so this prevents that from happening